README.rst

```diff
@@ -10,26 +10,74 @@ yourself duplicating quite a bit of code around handling raw environment
 variables.
 
 ``envparse`` aims to eliminate this duplicated, often inconsistent parsing
-code and instead provide a single, easy-to-use wrapper that handles:
+code and instead provide a single, easy-to-use wrapper.
+
+Ideas, and code portions, have been taken from `django-environ
+<https://github.com/joke2k/django-environ>`_ project but made framework
+agnostic.
+
+
+Installing
+----------
+Through PyPI::
+
+    $ pip install envparse
+
+Manually::
+
+    $ git clone https://github.com/rconradharris/envparse && cd envparse
+    $ python setup.py install
+
+
+Usage
+-----
+In your settings or configuration module, first either import the standard
+parser or one with a schema::
+
+    # Standard
+    from envparse import env
+
+    # Schema
+    from envparse import Env
+    env = Env(BOOLEAN_VAR=bool, LIST_VAR=dict(type=list, subtype=int))
 
-* Casting environment variables to a type::
 
-    MAIL_ENABLED=1
+``env`` can then be called in two ways:
 
-    mail_enabled = env('MAIL_ENABLED', cast=bool)
-    if mail_enabled:
-        send_mail(...)
+* Type explicit: ``env('ENV_VAR_NAME', type=TYPE, ...)``
+* Type implicit (for Python builtin types only): ``env.TYPE('ENV_VAR_NAME', ...)``
+If type is not specified, explicitly or implicitly, then the default
+type is ``str``.
 
-* Specifying defaults::
 
-    max_rows = env('MAX_ROWS', cast=int, default=100)
-    rows = query(limit=max_rows)
+Casting to a specified type::
 
-* Proxying values, useful in Heroku for wiring up the environment
-  variables they provide to the ones that your app actually uses::
+    # Environment variable: MAIL_ENABLED=1
 
-    MAILGUN_SMTP_LOGIN=foo          # Heroku provides this with add-on
-    SMTP_LOGIN={MAILGUN_SMTP_LOGIN} # App uses proxied variable
+    mail_enabled = env('MAIL_ENABLED', type=bool)
+    # OR mail_enabled = env.bool('MAIL_ENABLED')
+    assert mail_enabled is True
+
+Casting nested types::
+
+    # Environment variable: FOO=1,2,3
+    foo = env('FOO'), subtype=int)
+    # OR: foo = env('FOO', type=list, subtype=int)
+    # Note that there is no way to implicitly call subtypes.
+    assert foo == [1, 2, 3]
+
+Specifying defaults::
+
+    # Environment variable MAX_ROWS has not been defined
+
+    max_rows = env.int('MAX_ROWS', default=100)
+    assert max_rows == 100
+
+Proxying values, useful in Heroku for wiring up the environment variables they
+provide to the ones that your app actually uses::
+
+    # Environment variables: MAILGUN_SMTP_LOGIN=foo,
+    # SMTP_LOGIN='$MAILGUN_SMTP_LOGIN'
```

Comment:
The dollar-sign form for proxied-values has returned :)

My problem with the dollar-sign, even though it makes the most intuitive sense, is that is a bit error-prone. If a user mistakenly leaves off single-quotes when doing a `heroku config:set`, then they'll end up setting their config-variable to an empty string.

Later on, when they try to figure out why their app is not working, they'll be presented with an empty-string and be thoroughly confused--as I was when this happened.

I decided to avoid the issue entirely and use braces so we avoid the issue with bash escaping entirely. It's slightly less intuitive in the beginning, but never having to worry about bash-escaping again is a Good Thing (tm)!

Thoughts here?